src/systemctl/systemctl-edit.c

```
@@ -317,12 +317,13 @@ int verb_edit(int argc, char *argv[], void *userdata) {
                 .marker_end = DROPIN_MARKER_END,
                 .remove_parent = !arg_full,
                 .overwrite_with_origin = true,
+                .stdin = arg_stdin,
         };
         _cleanup_strv_free_ char **names = NULL;
         sd_bus *bus;
         int r;
 
-        if (!on_tty())
+        if (!on_tty() && !arg_stdin)
```

Hmm, do we really need this as a new flag? Maybe just call `isatty(STDIN_FILENO)` somewhere in `edit-util.c` and drop all `on_tty()` checks? Then `machinectl` and `networkctl` can automatically gain the ability too.

---

I think this would be very prone to errors. Also, not all editors require stdin to be connected. In fact, I expect that most people nowadays use graphical editors that aren't connected to the terminal. Overall, I think this would make things confusing and error-prone.

---

> Also, not all editors require stdin to be connected. In fact, I expect that most people nowadays use graphical editors that aren't connected to the terminal. 

Hmm? I don't really follow? To invoke `whateverctl edit` currently, stdio must be connected to terminals anyway. I don't understand how graphical editors bypass this?

---

Are you concerned that people call this in detached scripts with `EDITOR=some-graphical-editor`? Currently, it fails hard. With the proposed change, it replaces the current file. Yeah, maybe this is a bit more confusing, but that's unsupported anyway.

---

Currently it fails, but I think we should leave the option open to handling this differently in the future. But anyway, I think it'd be to error-prone to try to guess the intent.

---

Also, this is primarily intended for scripting. I think it's more important to be clear in a script than to save a few characters.